node/lib/util/commit.js

```diff
@@ -1762,3 +1767,280 @@ exports.parseSplitCommitMessages = function (text) {
         subMessages: subMessages,
     };
 };
+
+function errorWithStatus(status, relCwd, message) {
+    throw new UserError(`\
+${PrintStatusUtil.printRepoStatus(status, relCwd)}
+${colors.yellow(message)}
+`);
+}
+
+function checkForPathIncompatibleSubmodules(repoStatus, relCwd) {
+
+    const Commit = require("../util/commit");
+
+    if (Commit.areSubmodulesIncompatibleWithPathCommits(repoStatus)) {
+            errorWithStatus(repoStatus, relCwd, `\
+Cannot use path-based commit on submodules with staged commits or
+configuration changes.`);
+    }
+}
+
+function abortForNoMessage() {
+    throw new UserError("Aborting commit due to empty commit message.");
+}
+
+/**
+ * Perform the commit command in the specified `repo`.  Use the specified `cwd`
```

Comment:
Can you clarify how `cwd` is used?  As in, "paths are displayed relative to `cwd`" or whatever.